ddmrp/models/stock_buffer.py

```diff
@@ -1461,30 +1465,46 @@ def action_view_supply_outside_dlt_window(self):
     @api.model
     def cron_ddmrp_adu(self, automatic=False):
         """calculate ADU for each DDMRP buffer. Called by cronjob."""
+        auto_commit = not getattr(threading.currentThread(), "testing", False)
         _logger.info("Start cron_ddmrp_adu.")
-        buffers = self.search([])
+        buffer_ids = self.search([]).ids
         i = 0
-        j = len(buffers)
-        for b in buffers:
-            try:
-                i += 1
-                _logger.debug("ddmrp cron_adu: {}. ({}/{})".format(b.name, i, j))
-                if automatic:
-                    with self.env.cr.savepoint():
+        j = len(buffer_ids)
+        for buffer_chunk_ids in split_every(self.CRON_DDMRP_CHUNKS, buffer_ids):
+            for b in self.browse(buffer_chunk_ids).exists():
```

Comment:
Any particular reason to add these `exists` calls?

Reply:
(in any case this is no blocking and I will continue merging, just willing to know if any problem you encountered forced the change or it was just kind of a good practice)

Reply:
We do an initial search to get all the buffer ids, and then we commit: on commit, the db transaction obtains a new snapshot, so if any record has been deleted in another transaction, we'll try to browse it and will fail to read the record when we iterate over it.

Reply:
Got it, thanks.